webrecorder/browsertrix
# feat: Enable self-service user access to billing portal

Resolves https://github.com/webrecorder/browsertrix/issues/1875
Follows https://github.com/webrecorder/browsertrix/pull/1914



### Changes

- When billing is enabled, adds billing tab to org settings that displays billing information if applicable
- Handles external link to manage plan
- Refactors org quota type to always be present
- Refactors org settings into `TailwindComponent`

### Manual testing

1. Log in as self-serve sign-up org admin
2. Click "Org Settings" > "Billing". Verify correct plan information is shown
3. Click "Manage Plan". Verify new tab is opened and redirects to billing portal
4. Log out
5. Log in as managed org admin
6. Click "Org Settings" > "Billing". Verify correct plan information is shown
7. Click "Contact Sales". Verify email client is opened with subject line set to "Pro plan change request (`org name`)"

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Org Settings - Billing (active subscription) | <img width="586" alt="Screenshot 2024-07-09 at 11 23 43 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/f82b8a2f-9435-4ed2-bf6f-f586952816c5"> |
| Org Settings - Billing (info text) | <img width="387" alt="Screenshot 2024-07-09 at 11 22 34 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/e7072d82-c8d0-4a62-8c2f-37f2ef0d49fb"> |
| Org Settings - Billing (paused subscription) | <img width="973" alt="Screenshot 2024-07-09 at 10 45 34 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/1a873df0-3825-4425-902b-b7af6bf50c9b"> |
| Org Settings - Billing (canceling subscription) | <img width="577" alt="Screenshot 2024-07-09 at 11 33 53 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/ce5953a0-8720-4869-b419-1c3dc69a81ca"> |
| Org Settings - Billing (canceled subscription) | <img width="984" alt="Screenshot 2024-07-01 at 8 00 49 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/700a3a99-6cb7-4491-ab48-6a53018c9ede"> |
| Org Settings - Billing (no subscription) | <img width="974" alt="Screenshot 2024-07-09 at 10 46 15 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/82ed9c73-dd53-4b73-b2c4-46fabf1a20cb"> |
| Billing Portal Redirect (loading) | <img width="290" alt="Screenshot 2024-07-09 at 10 57 28 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/e97fa4e7-ec19-4a1d-a2e2-1e2f37b28446"> |
| Billing Portal Redirect (no portal URL) | <img width="466" alt="Screenshot 2024-07-09 at 10 44 52 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/700739d7-814c-481d-86af-d98c332a8374"> |

### Follow-ups

The org subscription information could be stale after billing updates made in the external billing portal. It's probably ok that the user needs to reload the page, for now, but we may want to refresh org data when clicking through the settings tabs.
